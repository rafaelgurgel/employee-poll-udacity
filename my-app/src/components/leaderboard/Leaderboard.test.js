import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Leaderboard from './LeaderBoard'
import '@testing-library/jest-dom'

beforeAll(() => {
  jest.spyOn(console, 'warn').mockImplementation((message) => {
    if (!message.includes('React.startTransition')) {
      console.warn(message)
    }
  })
})

afterAll(() => {
  jest.restoreAllMocks()
})

const mockReducer = (state = {
  users: {
    sarahedo: {
      id: 'sarahedo',
      name: 'Sarah Edo',
      questions: ['q1', 'q2'],
      answers: { q3: 'optionOne' },
    },
    tylermcginnis: {
      id: 'tylermcginnis',
      name: 'Tyler McGinnis',
      questions: ['q3'],
      answers: { q1: 'optionTwo' },
    },
  },
}) => state

const mockStore = createStore(mockReducer)

describe('Leaderboard Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <Leaderboard />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
