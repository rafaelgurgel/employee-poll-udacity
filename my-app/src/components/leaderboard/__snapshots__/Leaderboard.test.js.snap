// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="leaderboard-container"
  >
    <h3
      class="leaderboard-title"
    >
      Leaderboard
    </h3>
    <ul
      class="leaderboard-list"
    >
      <li
        class="leaderboard-item"
      >
        <div
          class="leaderboard-item-content"
        >
          <img
            alt="Avatar of Sarah Edo"
            class="leaderboard-avatar"
            src="defaultAvatar.svg"
          />
          <div
            class="leaderboard-user-details"
          >
            <h4>
              Sarah Edo
            </h4>
            <p>
              Questions Asked: 2
            </p>
            <p>
              Questions Answered: 1
            </p>
            <p>
              Total: 3
            </p>
          </div>
        </div>
      </li>
      <li
        class="leaderboard-item"
      >
        <div
          class="leaderboard-item-content"
        >
          <img
            alt="Avatar of Tyler McGinnis"
            class="leaderboard-avatar"
            src="defaultAvatar.svg"
          />
          <div
            class="leaderboard-user-details"
          >
            <h4>
              Tyler McGinnis
            </h4>
            <p>
              Questions Asked: 1
            </p>
            <p>
              Questions Answered: 1
            </p>
            <p>
              Total: 2
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
